plugins/chat/app/jobs/regular/chat/notify_mentioned.rb

```diff
@@ -145,13 +145,39 @@ def process_mentions(user_ids, mention_type)
         memberships = get_memberships(user_ids)
 
         memberships.each do |membership|
-          mention = ::Chat::Mention.find_by(user: membership.user, chat_message: @chat_message)
+          mention = find_mention(mention_type, membership.user.id)
           if mention.present?
             create_notification!(membership, mention, mention_type)
             send_notifications(membership, mention_type)
           end
         end
       end
+
+      def find_mention(mention_type, user_id)
+        mention_klass = resolve_mention_klass(mention_type)
+
+        target_id = nil
+        if mention_klass == ::Chat::UserMention
+          target_id = user_id
+        elsif mention_klass == ::Chat::GroupMention
+          target_id = Group.where("LOWER(name) = ?", mention_type).first.id
+        end
+
+        mention_klass.find_by(chat_message: @chat_message, target_id: target_id)
```

Comment:
not sure I like to have `@chat_message` as an implicit dependency, would probably prefer this to be an argument of `find_mention`

Reply:
Fixed